willow/registry.py

```diff
@@ -34,6 +38,7 @@ def __init__(self):
         self._registered_operations = defaultdict(dict)
         self._registered_converters = {}
         self._registered_converter_costs = {}
+        self._registered_optimizers: Set["OptimizerBase"] = set()
```

Comment:
Do we not care about preserving the order? I suspect this could make for some weird inconsistencies between environments.

Reply:
fixed in 66678bb